### ISSUE TITLE
Copy Paste Behaviour like linux terminal 

### ISSUE THREAD

**Issue by jigneshthummar** [warpdotdev/Warp#2853]
### Describe the solution you'd like?

```text
i just wish default command line behaviour like on selection copy and on right click paste text is something if you can add that will help

ctrl+c and ctrl+v is even better which you guys have but its just my (and probably many others) muscle memory
```

### How important is this feature to you?

1 (Not too important)

### Warp Internal (ignore) - linear-label:770f6576-d6c0-4e4f-a259-fc64b5156087

None

**Comment by dannyneira** [comment]
Thanks for this feature request!

Please :+1:  and subscribe to this request as it helps us gauge interest. We will post any updates here.

**Comment by kali-lover333** [comment]
Earlier when i open terminal in kali the title bar of terminal shows kali now shell no 1 shows up.
And my username and machine name didn't show up, just a simple black interface when i press ctrl C
then the shell name change to kali@kali and i can see my username . but why plz anybuddy knows how
to fix the problem even i cant see my history and even i didnt delete my history.

**Comment by AnjaLiebermann** [comment]
I am just statring with Warp terminal and that this feature is missing is a killer for me. I guess I stay with iTerm as long as this isn't available.

**Comment by adrfantini** [comment]
Now that Warp for Linux is available, this is a must.

This is a feature that every single app on Linux has (with some very rare exceptions), and once you get used to it you cannot really live without it!

**Comment by igorsantos07** [comment]
Not sure if there's any terminal which pastes _directly_ from right-click, but I've always experienced pasting with _middle_ click on Linux. And that's _definitely_ missing.

**Comment by AnjaLiebermann** [comment]
> Not sure if there's any terminal which pastes _directly_ from right-click, but I've always experienced pasting with _middle_ click on Linux. And that's _definitely_ missing.
> 
> Other than that, this seems like a dupe of #2737, and maybe #416 as well.

Of course you are correct: Paste is the middle click!
And wehen you are used to it being available in a shell, then you really miss it!

**Comment by dannyneira** [comment]
Hi Folks, I know this doesn't solve the issue of right-click paste, but we've released middle-click paste support in today's release [v0.2024.03.12.08.02](https://docs.warp.dev/getting-started/changelog#id-2024.03.14-v0.2024.03.12.08.02).

**Comment by cgoerner** [comment]
Thanks @dannyneira,
Really appreciate this seemingly small but surprisingly important feature for those that use it.
Highlight to copy and middle-click to paste is working for me.

One tiny issue though. When you highlight some text from a block, the focus changes to that block, so before you can use middle-click to paste that text into the prompt, you first have to left-click to change the focus to the prompt. If the middle-click also change the focus, that would be nice.

**Comment by Edke** [comment]
Perfect, works .

 A minor comment: in generic terminals just selecting and right after middle works. In Warp you need to change focus on input area to be able to paste. But this is a minor issue and can live with this :)

**Comment by SilentGlasses** [comment]
@Edke 
>A minor comment: in generic terminals just selecting and right after middle works. In Warp you need to change focus on input area to be able to paste. But this is a minor issue and can live with this :)

This was fixed and released today, update your terminal and you will see it working now. See #4609 if you still have issues after the update.

**Comment by AnjaLiebermann** [comment]
I have the newest version of Warp and will try to use it now that  copy & paste with the middle click should work. Do I have to change any setting to make it work because so far nothing pastes with the middle click? 
Mind I am a new user to warp because without the middle click I didn't bother to try it out.

**Comment by Edke** [comment]
> This was fixed and released today, update your terminal and you will see it working now. See #4609 if you still have issues after the update.

Confirming at least for me, it works as expected. Just an left click or drag and select and copy in any of previous blocks and middle click right after will paste into prompt block, thank you !

**Comment by AnjaLiebermann** [comment]
![grafik](https://github.com/warpdotdev/Warp/assets/56544668/2703a12e-1e3d-4ca4-9e09-a8a50a35ec15)
here you see me trying to mark "webflex-1982" and instead "main" is marked. So not only I cannot paste I cannot even mark the text I want to paste. 
System info: Mac 13.6.3 (22G436), newest Warp, logitech mouse with emulated middle click

**Comment by vorporeal** [comment]
Hi @AnjaLiebermann - mind filing a separate issue for the fact that the selection is on the line, assuming it hasn't been fixed yet?

Going to close out this issue as completed now that middle-click paste is supported on Linux.

There were some other things discussed above that we have separate tracking issues for; please subscribe to and thumbs-up these if you're still waiting on them:
* https://github.com/warpdotdev/Warp/issues/416 - Support for middle-click paste on macOS
* https://github.com/warpdotdev/Warp/issues/4627 - Support for right-click paste
